### DISCUSSION TITLE
Ruff formatting error

### DISCUSSION THREAD

**Discussion by cs2018ncsa** [zed-industries/zed#19631] [Feature Requests]
I have no idea what's going on here. i think the issue started with the last update but i don't know how to confirm that. ruff no longer seems to be working in my environment. on startup, i see:
![image](https://github.com/user-attachments/assets/4a6c1936-8104-45f4-b0ac-5b57a64a2086)

when i click to show error:

```
Language server error: ruff

oneshot canceled
-- stderr--
```

Here are my settings:

```
   "languages": {
    "Python": {
      "language_servers": ["ruff", "pyright"],
      "format_on_save": "on",
      "formatter": [
        {
          "code_actions": {
            "source.organizeImports.ruff": true,
            "source.fixAll.ruff": true
          }
        },
        {
          "language_server": { "name": "ruff" }
        }
      ]
    }
  },
  "lsp": {
    "ruff": {
      "initialization_options": {
        "settings": {
          // Ruff server settings goes here
          // "lineLength": 80,
          "lint": {
            "extendSelect": ["I"]
          }
        }
      }
    }
  },
```

any help?

**Comment by cs2018ncsa** [comment]
and now i'm getting an error on autoupdate:
![image](https://github.com/user-attachments/assets/3d7d0468-a1a3-4ad4-8ba1-c501e87b6204)

**Comment by cs2018ncsa** [comment]
i'm still not sure what the actual problem is but it has something to do with ruff being installed in the project environment i'm running zed from. i'm using poetry and when i remove ruff via `poetry remove ruff` everything works fine and i get formatting on save.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
